indico/modules/events/management/controllers.py

```diff
@@ -328,6 +400,89 @@ def _process(self):
         return jsonify_data(flash=False)
 
 
+class RHClonePreview(RHManageEventBase):
+    def _process(self):
+        form = CloneRepeatabilityForm()
+
+        if form.repeatability.data == 'interval':
+            clone_calculator = IntervalCloneCalculator(self.event_new)
+        elif form.repeatability.data == 'pattern':
+            clone_calculator = PatternCloneCalculator(self.event_new)
+        else:
+            raise BadRequest
+
+        try:
+            dates = clone_calculator.calculate(request.form)
```

Comment:
can this one also raise a `ValueError`? if not the exception handling here is a bit overkill